app.js

```diff
@@ -231,6 +315,77 @@ function checkNodes() {
 setInterval(checkNodes, 60*1000);
 checkNodes();
 
+function getAccounts() {
+  // gets accounts from core and stores it in redis
+  axios.get(MONITOR_URL+'accounts').then(function(resp){
+    var day = moment();
+    let multi = redisClient.multi();
+    var key = "accounts_"+day.format("YYYY-MM-DD")
+    multi.set(key, resp.data.count);
+    // Expire the key after 90 days
+    multi.expire(key, 60*60*24*90);
+
+    multi.exec(function (err, res) {
+      if (err) {
+        console.error(err);
+        return;
+      }
+      console.log("Added total accounts stats: "+key+" "+res);
+    });
+  });
+}
+setInterval(getAccounts, 3600000);
+getAccounts();
+
+
+function getAssets() {
+  // gets assets from core and stores it in redis
+  axios.get(MONITOR_URL+'assets').then(function(resp){
+    var day = moment();
+    let multi = redisClient.multi();
+    var key = "assets_"+day.format("YYYY-MM-DD")
+
+    multi.set(key, resp.data.count);
+    // Expire the key after 90 days
+    multi.expire(key, 60*60*24*90);
+
+    multi.exec(function (err, res) {
+      if (err) {
+        console.error(err);
+        return;
+      }
+      console.log("Added total assets stats: "+key+" "+res);
+    });
+  });
+}
+setInterval(getAssets, 3600000);
+getAssets();
+
+function getDexVolume() {
+  // gets assets from core and stores it in redis
+  axios.get('http://ticker.stellar.org').then(function(resp){
+    var day = moment();
+    let multi = redisClient.multi();
+    var key = "dex_"+day.format("YYYY-MM-DD")
+
+    for (var i = 0; i < resp.data.length; i++) {
+      multi.hmset(key, resp.data[i].Name, JSON.stringify(resp.data[i]));
+    }
+    // Expire the key after 90 days
+    multi.expire(key, 60*60*24*90);
+
+    multi.exec(function (err, res) {
+      if (err) {
+        console.error(err);
+        return;
+      }
+      console.log("Added DEX volume stats: "+key+" "+res);
+    });
+  });
+}
+setInterval(getDexVolume, 3600000);
+getDexVolume();
+
```

Comment:
A lot of code is duplicated here so let's find a better design. Here's my idea how to solve it:
* Let's move all functions that get data from external services to another package (`data-providers`?). Let's call them `DataProviders`. So we will have: `AccountsDataProvider`, `AssetDataProvider`, `DEXDataProvider`.
* `Storage` class will be responsible for saving and loading data from Redis (gets `redis` object in contructor). It will have the following methods:
  * `updateAccountsCount(date, count)`
  * `updateAssetsCount(date, count)`
  * `updateOperationsCount(date, count)`
  * `updateTransactionsCount(date, count)`
  * `updateDEXVolume(date, asset, volume)`
  * ...other `update` methods.
  * `getAccountsCount(days)` - return number of accounts in the last `days` days
  * `getAssetsCount(days)`
  * `getOperationsCount(days)`
  * `getDEXVolume(days, asset)`
  * ...other `get` methods.
* Finally we will have `Collector` class that will import all data providers and call them periodically to get data. It will also accept `Storage` object in contructor to persist data.
* Let's move all controllers functions (`function(req, res)`) to `controllers` package. Controllers should use `Storage` to get data from Redis.

Let me know if you have questions or other idea how to design it.